tmol/score/disulfide/disulfide_energy_term.py

```diff
@@ -0,0 +1,83 @@
+import torch
+import numpy
+
+from ..energy_term import EnergyTerm
+
+from tmol.database import ParameterDatabase
+from tmol.score.disulfide.params import DisulfideGlobalParams
+from tmol.score.disulfide.disulfide_whole_pose_module import (
+    DisulfideWholePoseScoringModule
+)
+
+from tmol.chemical.restypes import RefinedResidueType
+from tmol.pose.packed_block_types import PackedBlockTypes
+from tmol.pose.pose_stack import PoseStack
+
+
+class DisulfideEnergyTerm(EnergyTerm):
+    device: torch.device  # = attr.ib()
+
+    def __init__(self, param_db: ParameterDatabase, device: torch.device):
+        super(DisulfideEnergyTerm, self).__init__(param_db=param_db, device=device)
+
+        self.global_params = DisulfideGlobalParams.from_database(
+            param_db.scoring.disulfide, device
+        )
+        self.device = device
+
+    @classmethod
+    def score_types(cls):
+        import tmol.score.terms.disulfide_creator
+
+        return tmol.score.terms.disulfide_creator.DisulfideTermCreator.score_types()
+
+    def n_bodies(self):
+        return 1
+
+    def setup_block_type(self, block_type: RefinedResidueType):
+        super(DisulfideEnergyTerm, self).setup_block_type(block_type)
+
+        if hasattr(block_type, "disulfide_connections"):
+            return
+
+        disulfide_connections = numpy.array([], dtype=numpy.int32)
+        if "dslf" in block_type.connection_to_cidx.keys():
+            disulfide_connections = numpy.append(
+                disulfide_connections, [block_type.connection_to_cidx["dslf"]]
+            )
+
+        setattr(block_type, "disulfide_connections", disulfide_connections)
+
+    def setup_packed_block_types(self, packed_block_types: PackedBlockTypes):
+        super(DisulfideEnergyTerm, self).setup_packed_block_types(packed_block_types)
+
+        if hasattr(packed_block_types, "disulfide_conns"):
+            return
```

Comment:
Something like this? https://github.com/uw-ipd/tmol/pull/254/commits/99023c48306a08c616d2f38c0071aae5c5ff2d37

It feels a little weird to be putting this in the packed block types, since it isn't a feature of the block types and this data won't change if the active block types change. We could consider putting it in DisulfideEnergyTerm::\_\_init\_\_ and have it done once right after it loads the params from the database.

Reply:
If I understood you correctly, something like this? https://github.com/uw-ipd/tmol/pull/254/commits/17390b62a134432186c2f4dfcfcbc04c1caa9204

Should I be doing this test for all cached data members?